ai_genomics/analysis/tag_evaluation/make_tag_validation_dataset.py

```diff
@@ -0,0 +1,130 @@
+"""Script to 1) extract openalex concepts from openalex sample and
+2) generate validation dataset to compare tags across
+sample of sample of texts with DBpedia and OpenAlex tags."""
+
+from ai_genomics.getters.data_getters import load_s3_data, get_s3_dir_files, save_to_s3
+
+from ai_genomics import bucket_name, logger
+from typing import List, Union
+import pandas as pd
+import requests
+
+SAMPLES_DIR = get_s3_dir_files(bucket_name, "inputs/ai_genomics_samples/")
+OA_SAMPLE = load_s3_data(
+    bucket_name, "inputs/ai_genomics_samples/ai_genomics_openalex_samples.csv"
+)
+
+
+def make_openalex_api_url(work_url: str) -> str:
+    """Convert openalex sample work id into json-compatible url."""
+    return f"https://api.openalex.org/works/{work_url.split('/')[-1]}"
+
+
+def get_openalex_sample_concepts(oa_sample: pd.DataFrame = OA_SAMPLE) -> pd.DataFrame:
+    """Calls OpenAlex API to get concepts per OpenAlex sample work id.
+        Returns OpenAlex sample with associated concepts, concept
+        scores and concept levels.
+
+    Args:
+        oa_sample: DataFrame of OpenAlex text samples
+    """
+    oa_sample["api_url"] = oa_sample["work_id"].apply(make_openalex_api_url)
+
+    oa_tags = []
+    for sample in list(oa_sample["api_url"]):
+        oa_response = requests.get(sample)
+        if oa_response.ok:
+            if oa_response.headers["Content-Type"] == "application/json":
+                oa_data = oa_response.json()
+                if "concepts" in oa_data.keys():
+                    tags = [oa["display_name"] for oa in oa_data["concepts"]]
+                    scores = [oa["score"] for oa in oa_data["concepts"]]
+                    levels = [oa["level"] for oa in oa_data["concepts"]]
+                    oa_tags.append([sample, tags, scores, levels])
+                else:
+                    logger.info(f"no concepts found for {sample}")
+            else:
+                logger.info("response is not in json")
+        else:
+            logger.info(f"response code not 200 - {oa_response}")
+
+    oa_with_concepts = pd.DataFrame(
+        oa_tags, columns=["api_url", "tags", "scores", "levels"]
+    )
+    oa_sample_with_concepts = pd.merge(oa_with_concepts, oa_sample, on="api_url")
+
+    return oa_sample_with_concepts[
+        ["work_id", "abstract_text", "tags", "scores", "levels"]
+    ]
+
+
+def load_sample_tags(
+    tags_dir: str = "oa_tags", sample_dirs: List[str] = SAMPLES_DIR
+) -> pd.DataFrame:
+    """Load and concatenate dataset samples with tags extracted.
+        returns a standardised dataframe of descriptions,
+        tags extracted and the data source
+    Args:
+        tags_dir: the tag method type as described in file names
+        sample_dirs: A list of file names in S3 dataset samples directory
+    """
+    dfs = []
+    for sample_dir in sample_dirs:
+        if tags_dir in sample_dir:
+            if sample_dir.endswith(".csv"):
```

Comment:
```suggestion
        if (tags_dir in sample_dir) and sample_dir.endswith(".csv"):
```
then unindent all below